deflate_rle.c

```diff
@@ -36,19 +35,11 @@ Z_INTERNAL block_state deflate_rle(deflate_state *s, int flush) {
         /* See how many times the previous byte repeats */
         if (s->lookahead >= STD_MIN_MATCH && s->strstart > 0) {
             scan = s->window + s->strstart - 1;
-            prev = *scan;
-            if (prev == *++scan && prev == *++scan && prev == *++scan) {
-                strend = s->window + s->strstart + STD_MAX_MATCH;
-                do {
-                } while (prev == *++scan && prev == *++scan &&
-                         prev == *++scan && prev == *++scan &&
-                         prev == *++scan && prev == *++scan &&
-                         prev == *++scan && prev == *++scan &&
-                         scan < strend);
-                match_len = STD_MAX_MATCH - (unsigned int)(strend - scan);
-                match_len = MIN(match_len, s->lookahead);
-            }
-            Assert(scan <= s->window + s->window_size - 1, "wild scan");
+            if (scan[0] == scan[1] && scan[1] == scan[2])
+                match_len = functable.compare256_rle(scan, scan+3)+2;
```

Comment:
Is that the single point of calling ``functable.compare256_rle``?

If yes, then maybe make the functions ``inline`` and instead of calling through a pointer in the ``functable``, use direct name substitution (maybe by macros) at this line?

Reply:
Good point. `functable.compare256` is only exposed in `functable` because of its use in `deflate_quick`.

Reply:
Not ``functable.compare256`` because ``functable.compare256`` depends on CPU features:

https://github.com/zlib-ng/zlib-ng/blob/c72cd309cafae5d7e5eed42b9c4f36a53414362d/functable.c#L312-L323

It's about ``functable.compare256_rle``, because it depends only on macro values.

Reply:
I mentioned `compare256` because I remembered why that one was in `functable` and `compare256_rle` should not be.

Reply:
Anyways, I have made some changes if you could review.